src/clj/game/cards.clj

```diff
@@ -207,6 +207,42 @@
                                                (vals selected-cards)))]
                            (effect-completed state side (make-result eid {:number counter-count :msg msg})))))}))
 
+(defn pick-credit-providing-cards
+  "Similar to pick-virus-counters-to-spend. Works on :recurring and normal credits."
+  ([provider-func] (pick-credit-providing-cards (hash-map) 0 nil provider-func))
+  ([target-count provider-func] (pick-credit-providing-cards (hash-map) 0 target-count provider-func))
+  ([selected-cards counter-count target-count provider-func]
```

Comment:
Can you order these so you're building them one at a time left to right? So `[provider-func]` then `[provider-func target-count]` then `[provider-func target-count selected-cards counter-count]`? It's a small thing, but it helps readability to have a given argument in the same position across multiple arities.

Reply:
Looks like the virus picker does the same thing. Feel like fixing that up too? :-P

Reply:
Yeah I've just mirrored what I've seen in the virus picker, I guess. I'll change the credit picker, but I'm still hoping to be able to merge the two pickers in the end...

Reply:
All hope is lost. This will never merge 😂 

I'll be refactoring the virus picker.